README.md

```diff
@@ -92,8 +121,172 @@ export class MyComponent implements OnInit {
+import {AppState} from '../states/exmaple.models';
+
+@Component({
+  selector: 'example-component',
+  templateUrl: 'example-component.html',
+})
+export class ExampleComponent implements OnDestroy {
+
+  ngDestroy = new Subject<void>();
+
+  constructor(
+    public exampleFormService: ExampleFormService,
+    private store: Store<AppState>,
+  ) { }
+
+  sendForm() {
+    this.store.dispatch(new ExampleStart(this.exampleFormService.form.value));
+  }
+
+  ngOnDestroy() {
+    this.ngDestroy.next();
+    this.ngDestroy.complete();
+  }
+}
+```
+
+##### Effect (generated)
+- the dispatched method is intercepted by an effect which calls the API via the generated API service
+- upon success, a `Success` action is dispatched (payload is the server response data)
+- upon error, an `Error` action is dispatched (payload is the error message sent from the server)
+
+```typescript
+@Injectable()
+export class ExampleEffects {
+  @Effect()
+  CreateProductCategory = this.storeActions.ofType<actions.Start>(actions.Actions.START).pipe(
+    switchMap((action: actions.Start) => this.exampleService.exampleEndpointMethod(action.payload)
+      .pipe(
+        map(result => new actions.Success(result)),
+        catchError((error: HttpErrorResponse) => of(new actions.Error(error.message))),
+      ),
+    ),
+  );
+
+  constructor(
+    private storeActions: Actions,
+    private adminproductService: AdminProductService,
+  ) {}
+}
+```
+
+##### Reducer (generated)
+- the reducer catches the `Success` / `Error` actions dispatched by the generated effect and stores the payloads to the store
+```typescript
+export interface ExampleState {
+  data: __model.ExampleServerResponseInterface;
+  loading: boolean;
+  error: string;
+}
+
+export const initialExampleState: ExampleState = {
+  data: null,
+  loading: false,
+  error: null,
+};
+
+export const selectorName = 'Example';
+export const getExampleSelector = createFeatureSelector<ExampleState>(selectorName);
+
+export function ExampleReducer(
+  state: ExampleState = initialExampleState,
+  action: actions.ExampleAction): ExampleState {
+  switch (action.type) {
+    case actions.Actions.START: return {...state, loading: true, error: null};
+    case actions.Actions.SUCCESS: return {...state, data: action.payload, loading: false};
+    case actions.Actions.ERROR: return {...state, error: action.payload, loading: false};
+    default: return state;
+  }
+}
+```
+
+##### Component (again)
+- the data can be retrieved by subscribing to the store
+
+```typescript
+ngOnInit() {
+    this.exampleState = this.store.select(getExampleSelector)
+    // OR
+    this.data = this.store.select(s => ExampleState.data)
+    this.loading = this.store.select(s => ExampleState.loading)
+    this.error = this.store.select(s => ExampleState.error)
+}
+```
+
 ## Assumptions / limitations
 
+
 1. swagger file is in [version 2](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md) format, it must be json
```

Comment:
Please add an info, that we are skipping array-like structures, so only empty `FormArray` is created in generated forms.

Reply:
done